cozo-core/src/storage/newrocks.rs

```diff
@@ -0,0 +1,535 @@
+        let temp_dir = TempDir::new().into_diagnostic()?;
+        let db = new_cozo_newrocksdb(temp_dir.path())?;
+        
+        // Create test tables with proper ScriptMutability parameter
+        db.run_script(
+            r#"
+            {:create plain {k: Int => v}}
+            {:create tt_test {k: Int, vld: Validity => v}}
+            "#,
+            Default::default(),
+            ScriptMutability::Mutable,
+        )?;
+
+        Ok((temp_dir, db))
+    }
+
+    #[test]
+    fn test_basic_operations() -> Result<()> {
+        let (_temp_dir, db) = setup_test_db()?;
+
+        // Test data insertion
+        let mut to_import = BTreeMap::new();
+        to_import.insert(
+            "plain".to_string(),
+            crate::NamedRows {
+                headers: vec!["k".to_string(), "v".to_string()],
+                rows: (0..100)
+                    .map(|i| vec![DataValue::from(i), DataValue::from(i * 2)])
+                    .collect(),
+                next: None,
+            },
+        );
+        db.import_relations(to_import)?;
+
+        // Test simple query with ScriptMutability parameter
+        let result = db.run_script(
+            "?[v] := *plain{k: 5, v}",
+            Default::default(),
+            ScriptMutability::Immutable,
+        )?;
+        
+        assert_eq!(result.rows.len(), 1);
+        assert_eq!(result.rows[0][0], DataValue::from(10));
+
+        Ok(())
+    }
+    #[test]
+    fn test_time_travel() -> Result<()> {
+        let (_temp_dir, db) = setup_test_db()?;
+
+        // Insert time travel data
+        let mut to_import = BTreeMap::new();
+        to_import.insert(
+            "tt_test".to_string(),
+            crate::NamedRows {
+                headers: vec!["k".to_string(), "vld".to_string(), "v".to_string()],
+                rows: vec![
+                    vec![
+                        DataValue::from(1),
+                        DataValue::Validity(Validity::from((0, true))),
+                        DataValue::from(100),
+                    ],
+                    vec![
+                        DataValue::from(1),
+                        DataValue::Validity(Validity::from((1, true))),
+                        DataValue::from(200),
+                    ],
+                ],
+                next: None,
+            },
+        );
+        db.import_relations(to_import)?;
+
+        // Query at different timestamps
+        let result = db.run_script(
+            "?[v] := *tt_test{k: 1, v @ 0}",
+            Default::default(),
+            ScriptMutability::Immutable,
+        )?;
+        assert_eq!(result.rows[0][0], DataValue::from(100));
+
+        let result = db.run_script(
+            "?[v] := *tt_test{k: 1, v @ 1}",
+            Default::default(),
+            ScriptMutability::Immutable,
+        )?;
+        assert_eq!(result.rows[0][0], DataValue::from(200));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_range_operations() -> Result<()> {
```

Comment:
This test does not work for the old rocks implementation either. The problem is it is attempting to seek for the key with values (appended to the keys) included in the encoded prefix, which causes it to skip the first result. I can put a PR up with a fix, but it is unrelated to your implementation here.

Reply:
Gotcha